sdk/ai/Azure.AI.Projects/src/Custom/Agent/AgentsClient.Streaming.cs

```diff
@@ -0,0 +1,202 @@
+    /// <param name="cancellationToken"> The cancellation token to use. </param>
+    /// <exception cref="ArgumentNullException"> <paramref name="threadId"/> or <paramref name="assistantId"/> is null. </exception>
+    /// <exception cref="ArgumentException"> <paramref name="threadId"/> is an empty string, and was expected to be non-empty. </exception>
+#pragma warning disable AZC0015 // Unexpected client method return type.
+    public virtual CollectionResult<StreamingUpdate> CreateRunStreaming(string threadId, string assistantId, string overrideModelName = null, string overrideInstructions = null, string additionalInstructions = null, IEnumerable<ThreadMessage> additionalMessages = null, IEnumerable<ToolDefinition> overrideTools = null, float? temperature = null, float? topP = null, int? maxPromptTokens = null, int? maxCompletionTokens = null, TruncationObject truncationStrategy = null, BinaryData toolChoice = null, BinaryData responseFormat = null, IReadOnlyDictionary<string, string> metadata = null, CancellationToken cancellationToken = default)
+#pragma warning restore AZC0015 // Unexpected client method return type.
+    {
+        Argument.AssertNotNullOrEmpty(threadId, nameof(threadId));
+        Argument.AssertNotNull(assistantId, nameof(assistantId));
+
+        CreateRunRequest createRunRequest = new CreateRunRequest(
+            assistantId,
+            overrideModelName,
+            overrideInstructions,
+            additionalInstructions,
+            additionalMessages?.ToList() as IReadOnlyList<ThreadMessage> ?? new ChangeTrackingList<ThreadMessage>(),
+            overrideTools?.ToList() as IReadOnlyList<ToolDefinition> ?? new ChangeTrackingList<ToolDefinition>(),
+            stream: true,
+            temperature,
+            topP,
+            maxPromptTokens,
+            maxCompletionTokens,
+            truncationStrategy,
+            toolChoice,
+            responseFormat,
+            metadata ?? new ChangeTrackingDictionary<string, string>(),
+            null);
+        RequestContext context = FromCancellationToken(cancellationToken);
+
+        Response sendRequest() => CreateRunStreaming(threadId, createRunRequest.ToRequestContent(), context);
+        return new StreamingUpdateCollection(sendRequest, cancellationToken);
+    }
+
+    internal async Task<Response> CreateRunStreamingAsync(string threadId, RequestContent content, RequestContext context = null)
+    {
+        Argument.AssertNotNullOrEmpty(threadId, nameof(threadId));
+        Argument.AssertNotNull(content, nameof(content));
+
+        using var scope = ClientDiagnostics.CreateScope("AgentsClient.CreateRun");
+        scope.Start();
+        try
+        {
+            using HttpMessage message = CreateCreateRunRequest(threadId, content, context);
+            message.BufferResponse = false;
```

Comment:
Only these internal protocol method variants set message.BufferReponse now.

This means that any other convenience methods calling the public protocol method won't have to do something special to ensure the response is disposed to prevent leaking a network stream.

It also means that the protocol methods that return a response with an unbuffered stream aren't public, so we don't need to document the need to dispose the response to users.